components/public-api-server/pkg/proxy/errors.go

```diff
@@ -43,15 +43,28 @@ func categorizeRPCError(err error) *connect.Error {
 		// components/gitpod-protocol/src/messaging/error.ts
 		case 409:
 			return connect.NewError(connect.CodeAlreadyExists, fmt.Errorf(rpcErr.Message))
+		// components/gitpod-protocol/src/messaging/error.ts - SETUP_REQUIRED (no user)
+		case 410:
+			return connect.NewError(connect.CodeUnauthenticated, fmt.Errorf(rpcErr.Message))
+		// components/gitpod-protocol/src/messaging/error.ts - NEEDS_VERIFICATION (no user)
+		case 411:
+			return connect.NewError(connect.CodePermissionDenied, fmt.Errorf(rpcErr.Message))
+		case 429:
+			return connect.NewError(connect.CodeResourceExhausted, fmt.Errorf(rpcErr.Message))
 		// components/gitpod-messagebus/src/jsonrpc-server.ts#47
 		case -32603:
 			return connect.NewError(connect.CodeInternal, fmt.Errorf(rpcErr.Message))
 		case 470:
 			return connect.NewError(connect.CodePermissionDenied, fmt.Errorf(rpcErr.Message))
-
-		default:
-			return connect.NewError(connect.CodeInternal, fmt.Errorf(rpcErr.Message))
+		// components/gitpod-protocol/src/messaging/error.ts - INVALID_VALUE (no user)
+		case 650:
+			return connect.NewError(connect.CodeInvalidArgument, fmt.Errorf(rpcErr.Message))
+		}
+		// components/gitpod-protocol/src/messaging/error.ts - user errors
+		if rpcErr.Code >= 400 && rpcErr.Code < 500 {
+			return connect.NewError(connect.CodeInvalidArgument, fmt.Errorf(rpcErr.Message))
 		}
+		return connect.NewError(connect.CodeInternal, fmt.Errorf(rpcErr.Message))
```

Comment:
Could you please extend the tests for these cases?
[nit] Could you move the -32603 case either first or last, for my own ocd? :P